CSharp/DevVmPowershell/Helpers/Implementations/ProcessingHelper.cs

```diff
@@ -782,117 +427,385 @@ public async Task<bool> RemoveWorkerManagerServerFromDefaultResourcePoolAsync()
-							{
-								ArtifactID = workerServer.ArtifactID,
-								Name = workerServer.Name,
-								ServerType = new ResourceServerTypeRef()
-								{
-									ArtifactID = workerServer.ServerType.ArtifactID,
-									Name = workerServer.ServerType.Name
-								}
-							};
-
-							await resourcePoolManager.RemoveServerAsync(workerServerRef, defaultPoolRef);
-
-							resultServers = await resourcePoolManager.RetrieveResourceServersAsync(defaultPoolRef);
-
-							if (!resultServers.Exists(x => x.ServerType.Name.Equals(Constants.Processing.WorkerServer, StringComparison.OrdinalIgnoreCase)))
-							{
-								wasRemoved = true;
-								Console.WriteLine($"{nameof(RemoveWorkerServerFromDefaultResourcePoolAsync)} - Successfully removed Worker Server from Default Resource Pool");
-							}
-							else
-							{
-								wasRemoved = false;
-								Console.WriteLine($"{nameof(RemoveWorkerServerFromDefaultResourcePoolAsync)} - Failed to remove Worker Server from Default Resource Pool");
-							}
-
-						}
-						else
-						{
-							wasRemoved = true;
-							Console.WriteLine($"{nameof(RemoveWorkerServerFromDefaultResourcePoolAsync)} - Failed to remove Worker Server from Default Resource Pool as the Worker Server does not exist");
-						}
+						ArtifactID = serverTypeArtifactId
 					}
-					else
+				},
+				resourcePool = new
+				{
+					ArtifactID = defaultResourcePoolArtifactId
+				}
+			};
+			string addResourceServer = JsonConvert.SerializeObject(addResourceServerPayload);
+			HttpResponseMessage addServerResponse = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.ResourcePool.AddServerEndpointUrl, addResourceServer);
+			if (!addServerResponse.IsSuccessStatusCode)
+			{
+				throw new System.Exception("Failed to add the Server to the Default Resource Pool");
+			}
+		}
+
+		private async Task RemoveServerFromDefaultResourcePoolAsync(HttpClient httpClient, int serverArtifactId, int serverTypeArtifactId, int defaultResourcePoolArtifactId)
+		{
+			var removeServerResourcePoolPayload = new
+			{
+				resourceServer = new
+				{
+					ArtifactID = serverArtifactId,
+					ServerType = new
 					{
-						wasRemoved = true;
-						Console.WriteLine($"{nameof(RemoveWorkerServerFromDefaultResourcePoolAsync)} - Failed to remove Worker Server from Default Resource Pool as it already is not within the pool");
+						ArtifactID = serverTypeArtifactId
 					}
-
+				},
+				resourcePool = new
+				{
+					ArtifactID = defaultResourcePoolArtifactId
 				}
+			};
+			string removeServer = JsonConvert.SerializeObject(removeServerResourcePoolPayload);
+			HttpResponseMessage removeServerResponse = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.ResourcePool.RemoveServerUrl, removeServer);
+			if (!removeServerResponse.IsSuccessStatusCode)
+			{
+				throw new System.Exception("Failed to remove the Server from the Default Resource Pool");

```

Comment:
Is there additional details in the HttpResponse regarding the actual error. We should probably put that in the Exception we are throwing which will be helpful when troubleshooting.

Reply:
Added failure message